platform-sdk/swirlds-base/src/main/java/com/swirlds/base/utility/ToStringBuilder.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.base.utility;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A builder for toString methods.
+ * <p>
+ * It can be used to build a toString method for an object by appending values to the builder and then calling
+ * {@link #toString()} to get the final string. This maintains a uniform toString format across the codebase.
+ * <p>
+ * If you are using IntelliJ, you can use the "Generate" menu to generate a toString method using this class by adding
+ * the following templates:
+ * <pre>
+ * public java.lang.String toString() {
+ *   return new com.swirlds.base.utility.ToStringBuilder(this)
+ *   #foreach ($member in $members)
+ *   .append("$member.name", $member.accessor)
+ *   #end
+ *   .toString();
+ * }
+ * </pre>
+ * and with {@code appendSuper}:
+ * <pre>
+ * public java.lang.String toString() {
+ *   return new com.swirlds.base.utility.ToStringBuilder(this)
+ *   .appendSuper(super.toString())
+ *   #foreach ($member in $members)
+ *   .append("$member.name", $member.accessor)
+ *   #end
+ *   .toString();
+ * }
+ * </pre>
+ */
+public class ToStringBuilder {
+    private static final Pattern SUPER_STRING_PATTERN = Pattern.compile("\\[(.*?)\\]");
+    private static final String PACKAGE_PREFIX_PATTERN = "^.*\\.";
+    private static final String NULL_STRING = "<null>";
+    private static final char PACKAGE_SEPARATOR = '.';
+    private static final char INNER_CLASS_SEPARATOR = '$';
+    private final StringBuilder builder;
+    private final int initLength;
+
+    /**
+     * Create a new ToStringBuilder for the given object.
+     *
+     * @param object the object to build a toString for
+     */
+    public ToStringBuilder(@NonNull final Object object) {
+        builder = new StringBuilder();
+
+        builder.append(formatClassName(object.getClass().getName())).append("[");
+        initLength = builder.length();
```

Comment:
instead of using an int we could use a boolean or short for less memory usage. But that does not need to be done in this PR.